scene-lib/src/scenelib/annotations/el/AMethod.java

```diff
@@ -76,6 +101,98 @@ public String getMethodName() {
+    public void setTypeParameters(List<? extends TypeParameterElement> typeParameters) {
+        if (typeParameters == null) {
+            return;
+        }
+        if (this.typeParameters != null && !this.typeParameters.equals(typeParameters)) {
+            throw new Error(String.format("setTypeParameters(%s): already is %s%n",
+                                          typeParameters, this.typeParameters));
+        }
+        this.typeParameters = typeParameters;
+    }
+
+    /**
+     * Populates the method parameter map for the method. This is called from the constructor, so
+     * that the method parameter map always has an entry for each parameter.
+     *
+     * @param methodElt the method whose parameters should be vivified
+     */
+    public void vivifyAndAddTypeMirrorToParameters(ExecutableElement methodElt) {
+        for (int i = 0; i < methodElt.getParameters().size(); i++) {
+            VariableElement ve = methodElt.getParameters().get(i);
+            TypeMirror type = ve.asType();
+            Name name = ve.getSimpleName();
+            vivifyAndAddTypeMirrorToParameter(i, type, name);
+        }
+    }
+
+    /**
+     * Obtain the parameter at the given index, which can be further operated on to e.g. add a type
+     * annotation.
+     *
+     * @param i the parameter index (first parameter is zero)
+     * @param type the type of the parameter
+     * @param simpleName the name of the parameter
+     * @return an AFieldWrapper representing the parameter
+     */
+    public AField vivifyAndAddTypeMirrorToParameter(int i, TypeMirror type, Name simpleName) {
+        AField param = parameters.getVivify(i);
+        param.setName(simpleName.toString());
+        if (param.getTypeMirror() == null) {
+            param.setTypeMirror(type);
+        }
+        return param;
+    }
+
+    /**
+     * Get the return type.
+     *
+     * @return the return type, or null if the return type is unknown or void
+     */
+    public /*@Nullable*/ TypeMirror getReturnTypeMirror() {
+        return returnTypeMirror;
+    }
+
+    /**
+     * Set the return type.
+     *
+     * @param returnTypeMirror the return type
+     */
+    public void setReturnTypeMirror(/*@Nullable*/ TypeMirror returnTypeMirror) {
+        if (returnTypeMirror == null) {
+            return;
+        }
+        if (this.returnTypeMirror != null && this.returnTypeMirror != returnTypeMirror) {
+            throw new Error(String.format("setReturnTypeMirror(%s): already is %s%n",
+                                          returnTypeMirror, this.returnTypeMirror));
+        }
+        this.returnTypeMirror = returnTypeMirror;
+    }
+
+    /**
+     * Get the parameters, as a map from parameter index (0-indexed) to representation.
+     *
+     * @return an immutable copy of the vivified parameters, as a map from index to representation
+     */
+    public Map<Integer, AField> getParameters() {
+        return ImmutableMap.copyOf(parameters);
+    }
```

Comment:
Since `scenelib` generally doesn't use setter methods (and instead seems to only have public fields), does it make sense to add this getter? It's inconsistent with the existing code, but it is better practice.

Reply:
I agree.  I don't think it makes a big difference either way.  I left it this way to minimize changes in the client.

Reply:
That seems like a sensible reason.